server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -265,6 +265,10 @@ static boolean isParallelCollectionSupportedForResults(
         ToLongFunction<String> fieldCardinality,
         boolean isQueryPhaseParallelismEnabled
     ) {
+        // When profiling, we should never use more than one thread, none of the profiler objects are thread safe
+        if (source != null && source.profile()) {
+            return false;
```

Comment:
Should we update SearchSourceBuilder#supportsParallelCollection to not return false when profile is set as it's handled separately?

Reply:
Yeah, I can specify that and update the tests.